### PR TITLE
Latent codec: more natural initialization

### PR DESCRIPTION
**Pull request by YodaEmbedding**
- docs: updates for `sphinx-book-theme==1.0.0` (fixes warnings)
- latent codec: Use self-documenting keyword arguments instead of `_setdefault` et al.
- latent codec: Got rid of redundant `N` argument.

To do:

- [ ] Test everything again.

### CODE REVIEW

File: docs/source/latent_codecs.rst [hunk 1]
```diff
@@ -190,13 +190,10 @@ Using :py:class:`~compressai.models.base.SimpleVAECompressionModel`, some Google
...
-                N=N,
-
                 # A HyperpriorLatentCodec is made of "hyper" and "y" latent codecs.
                 latent_codec={
                     # Side-information branch with entropy bottleneck for "z":
                     "hyper": HyperLatentCodec(
-                        N,
                         h_a=h_a,
                         h_s=h_s,
                         entropy_bottleneck=EntropyBottleneck(N),
```
Comment (fracape) [inline docs/source/latent_codecs.rst:199]: entropy_bottleneck=EntropyBottleneck(**M**)  ? 

I see, for cheng /  sensetime, it's still N

File: docs/source/latent_codecs.rst [hunk 2]
```diff
@@ -220,13 +217,10 @@ Using :py:class:`~compressai.models.base.SimpleVAECompressionModel`, some Google
...
```
Comment (fracape) [inline docs/source/latent_codecs.rst:226]: entropy_bottleneck=EntropyBottleneck(**M**)  ?

File: compressai/latent_codecs/hyper.py [hunk 3]
```diff
@@ -68,13 +68,17 @@ class HyperLatentCodec(LatentCodec):
...
+        h_s: Optional[nn.Module] = None,
+        **kwargs,
+    ):
         super().__init__()
-        self._kwargs = kwargs
-        self.N = N
-        self._setdefault("entropy_bottleneck", lambda: EntropyBottleneck(N))
-        self._setdefault("h_a", nn.Identity)
-        self._setdefault("h_s", nn.Identity)
+        self.entropy_bottleneck = entropy_bottleneck or EntropyBottleneck(0)
```
Comment (fracape) [inline compressai/latent_codecs/hyper.py:79]: What is the rationale to init an entropybottleneck with 0 channels, can you extend it later?

In gainhyperlatentcodec, you init with 
self.entropy_bottleneck = entropy_bottleneck or EntropyBottleneck()

Comment (YodaEmbedding) [reply compressai/latent_codecs/hyper.py:79]: You're right; this was a bit silly. I wanted it to type-check. Possible alternatives:

1. Force callers to provide `entropy_bottleneck`:

```python
assert entropy_bottleneck is not None
self.entropy_bottleneck = entropy_bottleneck
```

2. Restore the `z_channels` argument:

```python
def __init__(
    self,
    z_channels: Optional[int] = None,
    entropy_bottleneck: Optional[EntropyBottleneck] = None,
):
    self.entropy_bottleneck = entropy_bottleneck or EntropyBottleneck(z_channels)
```

Whichever seems least confusing / easy to use.

**Review by fracape — APPROVED** [review]

**Pull request merged**